src/pages/nft/class/_classId/index.vue

```diff
@@ -573,6 +573,35 @@ export default {
       }
       return collectorsWithMemo;
     },
+    populatedBuyerWithMessage() {
+      if (!this.populatedDisplayEvents) {
+        return this.populatedCollectors;
+      }
+      const collectorsWithBuyerMessages = this.populatedCollectors.map(
+        buyer => {
+          const event = this.populatedDisplayEvents.find(
+            event => event.buyerMessage && buyer.id === event.toWallet
```

Comment:
I think we can make a filtered list `this.populatedDisplayEvents.filter(e => e.buyerMessage)` before the `.map` function and use that instead, for cleaner code and performance